vespa/application.py

```diff
@@ -213,14 +218,36 @@ def collect_training_data(
         :param number_additional_docs: Number of additional documents to retrieve for each relevant document.
         :param relevant_score: Score to assign to relevant documents. Default to 1.
         :param default_score: Score to assign to the additional documents that are not relevant. Default to 0.
+        :param show_progress: Prints the the current point being collected every `show_progress` step. Default to None,
+            in which case progress is not printed.
         :param kwargs: Extra keyword arguments to be included in the Vespa Query.
         :return: DataFrame containing document id (document_id), query id (query_id), scores (relevant)
             and vespa rank features returned by the Query model RankProfile used.
         """
 
         training_data = []
+        number_queries = len(labelled_data)
+        idx_total = 0
+        idx_query = 0
         for query_data in labelled_data:
```

Comment:
You can do the counting (both for queries and docs) by using a count iterator instead of keeping track your self.
```python
    for query_idx, query_data in zip(itertools.count(), labelled_data):
        ...
```

Reply:
And I see the even easier way is to use `enumerate`.
```python
   for query_idx, query_data in enumerate(labelled_data):
      ...
```

Reply:
True. Thanks.